cloudregister/switchcloudguestservices.py

```diff
@@ -0,0 +1,133 @@
+            sys.exit(1)
+        service_plugins = dict(
+            (os.path.basename(plugin), plugin) for plugin in glob.glob(
+                '/usr/lib/zypp/plugins/services/*')
+        )
+        for service in switch_services:
+            service_name = service.strip()
+            for activation in product_activations:
+                product_service = activation.get('service')
+                product_service_name = product_service.get('name')
+                if service_name == product_service_name:
+                    break
+            if not service_name:
+                continue
+            plugin = service_plugins.get(service_name)
+            if plugin:
+                log.info(
+                    '[Serviceswitch] Removing service plugin "%s"' % service_name
+                )
+                os.unlink(plugin)
+            cred_file = '/etc/zypp/credentials.d/%s' % service_name
+            if not os.path.exists(cred_file):
+                update_server = utils.get_smt()
+                user, password = utils.get_credentials(
+                    utils.get_credentials_file(update_server)
+                )
+                with open(cred_file, 'w') as creds:
+                    creds.write('username=%s\n' % user)
+                    creds.write('password=%s\n' % password)
+                os.chmod(cred_file, 0o600)
+            service_url = product_service.get('url')
+            cmd = [
+                'zypper',
+                '--non-interactive',
+                'addservice',
+                '--refresh',
+                service_url,
+                service_name
+            ]
+            # zypper is slow, wait loop
+            while utils.get_zypper_pid():
+                time.sleep(5)
+            result = utils.exec_subprocess(cmd)
+            if result:
+                log.info(
+                    '[Serviceswitch] Adding RIS for "%s" failed' % service_url
+                )
+                link_dest = os.path.join(
+                    '/usr/lib/zypp/plugins/services/',
+                    service_name
+                )
+                os.symlink('/usr/sbin/cloudguest-repo-service', link_dest)
+                log.info(
+                    '[Serviceswitch] Re-created service plugin "%s"' % service_name
+                )
+            else:
+                current_services.append(
+                    '/etc/zypp/services.d/%s.service' % service_name
+                )
+
+        all_services = glob.glob('/etc/zypp/services.d/*.service')
+        for service in all_services:
+            if service not in current_services:
+                os.unlink(service)
+        os.unlink(service_revert_file)
+        os.unlink("%s.lock" % service_revert_file)
+        # zypper is slow, wait loop
+        while utils.get_zypper_pid():
+            time.sleep(5)
+        utils.exec_subprocess(['zypper', '--non-interactive', 'refs', '-f'])
+    else:
+        utils.switch_services_to_plugin()
```

Comment:
Since we are doing major surgery we could have dropped this script at this point. We've not handed out `https://` from our update infrastructure in 5 or 6 years. If anyone still has an instance that old and has not update cloud-regionsrv-client yet access to the update infrastructure is probably broken anyway.